File: Rio/Rio.Web/Modules/ServerTypes/Workspace/SheetTypeForm.ts
import { StringEditor, TextAreaEditor, IntegerEditor, EnumEditor, ImageUploadEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { EPaperSize } from "./enums.EPaperSize";
import { initFormType } from "@serenity-is/corelib/q";

export interface SheetTypeForm {
    Name: StringEditor;
    Description: TextAreaEditor;
    TotalQuestions: IntegerEditor;
    EPaperSize: EnumEditor;
    HeightInPixel: IntegerEditor;
    WidthInPixel: IntegerEditor;
    SheetData: TextAreaEditor;
    SheetImage: ImageUploadEditor;
    OverlayImage: ImageUploadEditor;
    Synced: BooleanEditor;
    IsPrivate: BooleanEditor;
    SheetNumber: StringEditor;
    PdfTemplate: ImageUploadEditor;
}

export class SheetTypeForm extends PrefixedContext {
    static formKey = 'Workspace.SheetType';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SheetTypeForm.init)  {
            SheetTypeForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = IntegerEditor;
            var w3 = EnumEditor;
            var w4 = ImageUploadEditor;
            var w5 = BooleanEditor;

            initFormType(SheetTypeForm, [
                'Name', w0,
                'Description', w1,
                'TotalQuestions', w2,
                'EPaperSize', w3,
                'HeightInPixel', w2,
                'WidthInPixel', w2,
                'SheetData', w1,
                'SheetImage', w4,
                'OverlayImage', w4,
                'Synced', w5,
                'IsPrivate', w5,
                'SheetNumber', w0,
                'PdfTemplate', w4
            ]);
        }
    }
}

[EPaperSize]; // referenced types

File: Rio/Rio.Web/Modules/Workspace/SheetType/SheetTypeForm.cs
using Rio.Workspace.enums;
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.SheetType")]
    [BasedOnRow(typeof(SheetTypeRow), CheckNames = true)]
    public class SheetTypeForm
    {
        [HalfWidth]
        public string Name { get; set; }
        [TextAreaEditor(Rows =5)]
        public string Description { get; set; }
        [HalfWidth]
        public int TotalQuestions { get; set; }
        [HalfWidth]
        public EPaperSize EPaperSize { get; set; }
        [HalfWidth]
        public int HeightInPixel { get; set; }
        [HalfWidth]
        public int WidthInPixel { get; set; }
        [TextAreaEditor(Rows =5)]
        public string SheetData { get; set; }
        [HalfWidth]
        public string SheetImage { get; set; }
        [HalfWidth]
        public string OverlayImage { get; set; }
        [QuarterWidth]
        public bool Synced { get; set; }
/*        [QuarterWidth]
        public bool IsPublic { get; set; }*/
        [QuarterWidth]
        public bool IsPrivate { get; set; }
        [HalfWidth]
        public long SheetNumber { get; set; }
        public string PdfTemplate { get; set; }

        
        /*public DateTime InsertDate { get; set; }
        public int InsertUserId { get; set; }
        public DateTime UpdateDate { get; set; }
        public int UpdateUserId { get; set; }
        public short IsActive { get; set; }*/
    }
}